code/modules/mob/living/carbon/human/life/handle_mutations_and_radiation.dm

```diff
@@ -73,3 +97,201 @@
+					if(!(O.status & ORGAN_MUTATED|ORGAN_ROBOT))
+						candidates.Add(O)
+				if (candidates.len)
+					var/datum/organ/external/O = pick(candidates)
+					O.mutate()
+					to_chat(src, "<span class = 'notice'>Something is not right with your [O.display_name].</span>")
+			if(prob(5*rad_multiplier))
+				//Minor clone damage
+				adjustCloneLoss(rand(1,5))
+				to_chat(src, "<span class='warning'>[pick("You can feel your body becoming weak!</span>", \
+				"You feel like you're about to die!", \
+				"You feel every part of your body screaming in agony!", \
+				"A low, rolling pain passes through your body!", \
+				"Your body feels as if it's falling apart!", \
+				"You feel extremely weak!", \
+				"A sharp, deep pain bathes every inch of your body!")]")
+			if(prob(10)*minor_rad_multiplier)
+				//Blindness
+				var/datum/organ/internal/eyes/E = internal_organs_by_name["eyes"]
+				if(!E.robotic && !(sdisabilities & BLIND || disabilities & NEARSIGHTED))
+					to_chat(src, "<span class = 'danger'>[pick("Your eyesight starts to fade!","Your eyes go cloudy!","Are you going blind?")]</span>")
+					E.damage += 2.5
+					eye_blurry = min(eye_blurry+1.5,50)
+					if (E.damage >= E.min_broken_damage && !(sdisabilities & BLIND))
+						simple_message("<span class='warning'>You go blind!</span>","<span class='warning'>Somebody turns the lights off.</span>")
+						sdisabilities |= BLIND
+					else if (E.damage >= E.min_bruised_damage && !(disabilities & NEARSIGHTED))
+						simple_message("<span class='warning'>It becomes hard to see for some reason.</span>","<span class='warning'>Somebody turns the lights off.</span>")
+						eye_blind = 5
+						eye_blurry = 5
+						disabilities |= NEARSIGHTED
+						spawn(100)
+							disabilities &= ~NEARSIGHTED
+		if(rad_tick > RADDOSEDEADLY)
+			if(prob(2.5*major_rad_multiplier))
+				//Necrosis Have sloughs of meat fall off the subject, maybe a limb fall off or become skeletal. Look at frankenstein limb code
+				var/inst = pick(1, 2, 3)
+				switch(inst)
+
+					if(1)
+						//Drop some meat
+						to_chat(src, "<span class='warning'>A chunk of meat falls off of you!</span>")
+						var/totalslabs = 1
+						var/obj/item/weapon/reagent_containers/food/snacks/meat/allmeat[totalslabs]
+						var/sourcename = real_name
+						var/sourcejob = job
+						var/sourcenutriment = nutrition / 15
+						//var/sourcetotalreagents = mob.reagents.total_volume
+
+						for(var/i = 1 to totalslabs)
+							var/obj/item/weapon/reagent_containers/food/snacks/meat/human/newmeat = new
+							newmeat.name = sourcename + " " + newmeat.name
+							newmeat.subjectname = sourcename
+							newmeat.subjectjob = sourcejob
+							newmeat.reagents.add_reagent(NUTRIMENT, sourcenutriment / totalslabs) //Thehehe. Fat guys go first
+							//src.occupant.reagents.trans_to(newmeat, round (sourcetotalreagents / totalslabs, 1)) // Transfer all the reagents from the
+							allmeat[i] = newmeat
+
+							var/obj/item/meatslab = allmeat[i]
```

Comment:
- the allmeat list appears to be completely useless
- instead of creating a meatslab variable that holds the reference to the created meat, just use the already existing newmeat variable
- instead of forceMoving the new meat slab, do newmeat = new(get_turf(src))
- I believe getFromPool() already calls New()

In addition everything in this entire block of code is copypasted from either the gibber or the butcher knife (or maybe even both aaaaa) please make it a proc